### PR TITLE
Ci format

### PR DESCRIPTION
**Pull request by bwrsandman**
Using https://github.com/microsoft/snmalloc/blob/master/azure-pipelines.yml as a reference

### CODE REVIEW

File: azure-pipelines-ci.yml [hunk 1]
```diff
@@ -55,3 +55,43 @@ jobs:
...
+
+  steps:
+    - script: |
+        set -eo pipefail
+        wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+        sudo add-apt-repository "deb http://apt.llvm.org/xenial/ llvm-toolchain-xenial-8 main"
+        sudo apt-get update
+        sudo apt-get install -y clang-format-8 clang-tidy-8
+        sudo update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-8 100
+      displayName: 'Install clang'
```
Comment (bwrsandman) [inline azure-pipelines-ci.yml]: Would be better off if the docker had this installed already. @handsomematt

Comment (janisozaur) [reply azure-pipelines-ci.yml]: Angular has static builds of clang-format that get updated every so often: https://github.com/angular/clang-format

3-4MB binary is all you need. Keep in mind these are not directly tied to clang releases and _may_ produce slight variations in output in each version.

Comment (bwrsandman) [reply azure-pipelines-ci.yml]: Thanks. I'll try that. It does require npm though.

Comment (bwrsandman) [reply azure-pipelines-ci.yml]: Either @handsomematt has to install clang-format on the container or npm

Comment (janisozaur) [reply azure-pipelines-ci.yml]: The way I was using it was simply to download the binary directly from github, mark as a+x, then execute. No npm necessary really.

Comment (janisozaur) [reply azure-pipelines-ci.yml]: You can download whatever is on current master from this link: https://github.com/angular/clang-format/blob/master/bin/linux_x64/clang-format?raw=true

**Comment by handsomematt** [comment]
I'll add clang-format to the docker image itself - in the mean time we need some sort of ignore list for third party headers such as `Common/stb_image_write.h` would probably end up moving all that to a src/ThirdParty folder too.

**Comment by bwrsandman** [comment]
Good idea

**Comment by handsomematt** [comment]
https://github.com/openblack/compile-farm/commit/e408b85d3ad7452b53295a2ebe3c7d27c6867a2c

**Comment by bwrsandman** [comment]
This PR is mostly ready. I mentioned some changes to the formatting rules that I believe should be made (#208).

Before merging, the repo style should be fixed up (#207 and more) and the test run again to make sure nothing bad happens when publishing the diff with no errors.

**Comment by bwrsandman** [comment]
Requires #215 merged to have green.

**Comment by bwrsandman** [comment]
Rebased on the formatted code

**Pull request merged**